src/osdatahub/NamesAPI/names_api.py

```diff
@@ -0,0 +1,160 @@
+        key = environ.get("OS_API_KEY")
+        names = NamesAPI(key)
+        results = names.find("Buckingham Palace", limit=5)
+    """
+    __ENDPOINT = r"https://api.os.uk/search/names/v1/"
+    HEADERS = {"method": "POST",
+               "headers": "{'Content-Type': 'application/json'}"}
+
+    def __init__(self, key: str):
+        self.key = key
+
+    def __endpoint(self, api_name: str) -> str:
+        return self.__ENDPOINT + api_name + f"?key={self.key}"
+
+    @typechecked
+    def find(self, text: str, limit: int = 100,
+             bounds: Extent = None, bbox_filter: Extent = None,
+             local_type: Union[Iterable, str] = None) -> FeatureCollection:
+        """A free text query of the OS Names API
+
+        Args:
+            text (str): The free text search parameter
+            limit (int, optional): The maximum number of features to return.
+                Defaults to 100.
+            bounds (Extent, optional): Biases the results to a certain area. Must be British National Grid
+                (EPSG:27700) CRS
+            bbox_filter (Extent, optional): Filters the results to a certain area. Must be British National Grid
+                (EPSG:27700) CRS
+            local_type (Union[Iterable, str], optional): Filters the results to certain local types. Available local
+                types can be found at the bottom of https://osdatahub.os.uk/docs/names/technicalSpecification
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+        data = GrowList()
+        params = {"query": text}
+
+        if limit <= 0:
+            raise ValueError(f"Parameter \"limit\" must be a positive integer. Instead got {limit}")
+
+        if bounds:
+            if not bounds.crs == "EPSG:27700":
+                raise TypeError("Bounds must be in British National Grid CRS (EPSG:27700)")
+            params.update({"bounds": bounds.bbox.to_string()})
+        if bbox_filter or local_type:
+            if bbox_filter and (not bbox_filter.crs == "EPSG:27700"):
+                raise TypeError("Bounding Box filter must be in British National Grid CRS (EPSG:27700)")
+            params.update({"fq": self.__format_fq(bbox_filter, local_type)})
+
+        try:
+            n_required = min(limit, 100)
+            while n_required > 0 and data.grown:
+                params.update({"offset": len(data), "maxresults": n_required})
+                print(params)
+                response = requests.get(self.__endpoint("find"), params=params)
+                data.extend(self.__format_response(response))
+                n_required = min(100, limit - len(data))
+        except KeyError:
+            raise_http_error(response)
+        return addresses_to_geojson(data.values, "EPSG:27700")
+
+    def nearest(self, point: tuple, radius: float = 100, local_type: Union[Iterable, str] = None) -> FeatureCollection:
+        """Takes a pair of coordinates (X, Y)/(Lon, Lat) as an input
```

Comment:
Does the point have to be BNG? If so, might be good to remove the (lon, lat) in functions description